### PR TITLE
[stdlib] Make KeyValuePairs fully inlinable

### PR DESCRIPTION
**Pull request by airspeedswift**
The implementation is trivial and it has a significant benefit on performance.

Also gives KVP custom string conversions, factoring out the code used in `Dictionary`. And makes `_makeCollectionDescription` an extension method, cos it should be.

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Comment by airspeedswift** [comment]
@swift-ci please test

File: stdlib/public/core/KeyValuePairs.swift [hunk 1]
```diff
@@ -0,0 +1,141 @@
+//===--- KeyValuePairs.swift ----------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
```
Comment (lorentey) [inline stdlib/public/core/KeyValuePairs.swift]: 2018

File: stdlib/public/core/KeyValuePairs.swift [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+///     let runner = "Marlies Gohr"
+///     if let index = recordTimes.firstIndex(where: { $0.0 == runner }) {
+///         let time = recordTimes[index].1
+///         print("\(runner) set a 100m record of \(time) seconds.")
+///     } else {
+///         print("\(runner) couldn't be found in the records.")
+///     }
+///     // Prints "Marlies Gohr set a 100m record of 10.81 seconds."
+///
+/// Dictionary Literals as Function Parameters
```
Comment (lorentey) [inline stdlib/public/core/KeyValuePairs.swift]: I *think* this should read "Key-Value Pairs as Function Parameters", as the text talks about using `KeyValuePair` parameters with dictionary literals as arguments.

**Review by lorentey — APPROVED** [review]
The new description extensions are super neat!

File: stdlib/public/core/KeyValuePairs.swift (hunk 2 above)
Comment (airspeedswift) [reply stdlib/public/core/KeyValuePairs.swift]: Good spot. I'll do that in another PR and land this tho.

**Comment by airspeedswift** [comment]
@swift-ci please smoke test and merge

**Pull request merged**